File: types/tenantLlmConfig.ts
export type TenantLLMConfigSet = {
  api_key: string
  provider?: string
}

export type TenantLLMConfigRead = {
  id: string
  tenant_id: string
  provider: string
  last4: string
  is_active: boolean
  created_at: string
  updated_at: string | null
}

export type TenantLLMConfigStatus = {
  has_key: boolean
  provider: string
  last4: string | null
  is_active: boolean
}

File: backend/app/schemas/tenant_llm_config.py
from __future__ import annotations

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field


class TenantLLMConfigSet(BaseModel):
    """Payload for setting the OpenAI API key."""
    api_key: str = Field(..., min_length=10, max_length=512, description="OpenAI API key (sk-...)")
    provider: str = Field(default="openai", pattern=r"^[a-z0-9_-]+$", max_length=50)


class TenantLLMConfigRead(BaseModel):
    id: UUID
    tenant_id: UUID
    provider: str
    last4: str
    is_active: bool
    created_at: datetime
    updated_at: datetime | None = None

    model_config = {"from_attributes": True}


class TenantLLMConfigStatus(BaseModel):
    """Lightweight check: does the tenant have a key configured?"""
    has_key: bool
    provider: str = "openai"
    last4: str | None = None
    is_active: bool = False
